x-pack/filebeat/module/gcp/firewall/test/rare.log-expected.json

```diff
@@ -7,24 +7,24 @@
         "destination.port": 80,
         "event.action": "firewall-rule",
         "event.category": "network",
-        "event.dataset": "googlecloud.firewall",
+        "event.dataset": "gcp.firewall",
         "event.id": "1dobeotg13df9f5",
         "event.kind": "event",
-        "event.module": "googlecloud",
+        "event.module": "gcp",
         "event.type": [
             "connection",
             "denied"
         ],
         "fileset.name": "firewall",
-        "googlecloud.destination.instance.project_id": "local-test",
-        "googlecloud.destination.instance.region": "us-central1",
-        "googlecloud.destination.instance.zone": "us-central1-a",
-        "googlecloud.destination.vpc.project_id": "test-beats",
-        "googlecloud.destination.vpc.subnetwork_name": "mysubnet",
-        "googlecloud.destination.vpc.vpc_name": "default",
-        "googlecloud.firewall.rule_details.action": "DENY",
-        "googlecloud.firewall.rule_details.direction": "INGRESS",
-        "googlecloud.firewall.rule_details.ip_port_info": [
+        "gcp.destination.instance.project_id": "local-test",
+        "gcp.destination.instance.region": "us-central1",
+        "gcp.destination.instance.zone": "us-central1-a",
+        "gcp.destination.vpc.project_id": "test-beats",
+        "gcp.destination.vpc.subnetwork_name": "mysubnet",
+        "gcp.destination.vpc.vpc_name": "default",
+        "gcp.firewall.rule_details.action": "DENY",
+        "gcp.firewall.rule_details.direction": "INGRESS",
+        "gcp.firewall.rule_details.ip_port_info": [
```

Comment:
Are we backporting these changes in fields to 7.x? This would be breaking.

Reply:
@jsoriano Hmm good point. I was thinking if user continue using `googlecloud` module instead of `gcp`, then the output event will still have names `googlecloud.*`. Is there a better way for keeping this change backwards compatible? Maybe to use copy_fields processor?

Reply:
> I was thinking if user continue using googlecloud module instead of gcp, then the output event will still have names googlecloud.*.

Pipelines are shared, and they use the new name. Also dashboards, and they also use the new name.

> Is there a better way for keeping this change backwards compatible? Maybe to use alias?

We can go on with this as is, with `gcp` prefixes, but then we should add a breaking change note. The alternative would be to keep old field names, and rename them in 8.0.
Field aliases barely help with this, they would work for queries using these fields, but returned documents would only have the new fields, so for example dashboards wouldn't work.

Summarizing, I think we have these options:
* Going on with current changes renaming everything, but then we need to add breaking changes notes. We have to think if this is going to affect current user base and how. I am ok with going on with this if it looks good to you too.
* Rename the module, use module alias for backwards compatibility, but don't rename the fields till 8.0.
* Duplicate modules, so they can be used with any of the names, and remove the old one in 8.0. But this can be confusing and a maintenance burden.

The only time we have renamed modules it was with the apache module and it was on a major version, in 7.0, where we also renamed many fields for the migration to ECS.

Reply:
> Summarizing, I think we have these options:
> 
> * Going on with current changes renaming everything, but then we need to add breaking changes notes. We have to think if this is going to affect current user base and how. I am ok with going on with this if it looks good to you too.
> * Rename the module, use module alias for backwards compatibility, but don't rename the fields till 8.0.
> * Duplicate modules, so they can be used with any of the names, and remove the old one in 8.0. But this can be confusing and a maintenance burden.
> 
> The only time we have renamed modules it was with the apache module and it was on a major version, in 7.0, where we also renamed many fields for the migration to ECS.

@jsoriano Thank you so much for listing all the options. Filebeat `googlecloud` module is GA so I would avoid making it a breaking change in 7.x. This case, I prefer the second option: rename the module but keep the field names the same till 8.0. 

@andrewkroh WDYT?

Reply:
I think I like the second option too.

The field renames for 8.0 can happen now, they just should not be backported to 7.x.